### PR TITLE
BFZ API

### PR DESCRIPTION
**Pull request by eanders**

### CODE REVIEW

File: drivers/built_for_zero_report/app/models/built_for_zero_report/calculator.rb [hunk 1]
```diff
@@ -27,11 +27,63 @@ class Calculator
     # @param start_date [Date] The start of the reporting range
     # @param end_date [Date] The end of the reporting range
     # @param client_ids [Set, nil] If defined, limit the clients to the specified destination client ids
-    def initialize(cohort_key, start_date, end_date, client_ids: nil)
+    # @param client_ids [Array, nil] If defined, an array of string representations of the keys we will send to the API
```
Comment (awisema) [inline drivers/built_for_zero_report/app/models/built_for_zero_report/calculator.rb:30]:
```suggestion
    # @param data_keys [Array, nil] If defined, an array of string representations of the keys we will send to the API
```

File: drivers/built_for_zero_report/app/models/built_for_zero_report/calculator.rb [hunk 2]
```diff
@@ -27,11 +27,63 @@ class Calculator
...
+        'returned_from_housing' => returned_from_housing.count,
+        'returned_from_inactive' => returned_from_inactivity.count,
+        'name' => @user.name,
+        'email' => @user.email,
+        'organization' => @user.agency_name,
+        'date_interval_start' => @start_date.to_time.utc.strftime('%FT%T.000Z'), # match API format
+        'date_interval' => @interval,
+      }
+      # data['avg_lot_from_id_to_housing'] = average_lot_to_housing unless average_lot_to_housing.present?
+      # data.delete_if { |k, _| !@data_keys.include?(k) }
```
Comment (awisema) [inline drivers/built_for_zero_report/app/models/built_for_zero_report/calculator.rb]: Remove?

File: drivers/built_for_zero_report/app/views/built_for_zero_report/warehouse_reports/bfz/_filter.haml [hunk 3]
```diff
@@ -9,6 +9,9 @@
         = f.input :section, as: :select_two, collection: BuiltForZeroReport.sections, selected: @section_key
 
   - content_for :filter_actions do
-    = f.submit 'Update Report', class: ['btn', 'btn-primary'], data: { disable_with: false }
+    .d-flex.w-100
+      = f.submit 'Update Report', class: ['btn', 'btn-primary'], data: { disable_with: false }
+      .ml-auto
+        = f.submit 'Submit Report', class: ['btn', 'btn-secondary'], data: { disable_with: false, confirm: 'Submitting this report will send data for all sections to Built For Zero via their submission API. Are you sure you would like to proceed?' }
```
Comment (awisema) [inline drivers/built_for_zero_report/app/views/built_for_zero_report/warehouse_reports/bfz/_filter.haml:15]: Should this be commented out until we are ready to release?

**Review by awisema — CHANGES_REQUESTED** [review]

File: drivers/built_for_zero_report/app/views/built_for_zero_report/warehouse_reports/bfz/_filter.haml (hunk 3 above)
Comment (eanders) [reply drivers/built_for_zero_report/app/views/built_for_zero_report/warehouse_reports/bfz/_filter.haml:15]: Oh! Looks like I implemented the submit.  Should be good to leave this in.

**Pull request merged**